cJSON.c

```diff
@@ -1509,6 +1509,10 @@ static cJSON_bool parse_array(cJSON * const item, parse_buffer * const input_buf
 success:
     input_buffer->depth--;
 
+    if (head != NULL) {
```

Comment:
@miaoerduo thanks your suggestion! but this change make the testcase fail, I have to make some changes for it.

Reply:
Maybe the 0th is unnecessary? How about the others? I have test these in my project. :P

Reply:
No, it's needed. compiling it with option `cmake ENABLE_VALGRIND=On ..` when `make check`, memory check is very important.

Reply:
Sorry, I have commited with company's email.
So I have to changed the email in commits and force pushed again.

Reply:
May my modification be merged into this project? 
Is there any other thing I could do?
I haven't contributed to a project with so many stars!

Reply:
This version is released no more than 24 hours since you make this contribution, you have done a good job :), before I merge it, I need some time to make sure it won't introduce any problem.

Reply:
Wow! Thank YOU very much!